Userland/Libraries/LibCore/System/Promises.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/String.h>
+#include <AK/Vector.h>
+#include <Kernel/API/Pledge.h>
+#include <initializer_list>
+
+namespace Core::System {
+
+struct Promises {
+
+#define __ENUMERATE_PLEDGE_PROMISE(promise_name) bool promise_name : 1 { false };
+    ENUMERATE_PLEDGE_PROMISES
+#undef __ENUMERATE_PLEDGE_PROMISE
+
+    String to_string() const
+    {
+        Vector<String> promises_builder;
+
+#define __ENUMERATE_PLEDGE_PROMISE(promise_name) \
+    if (promise_name)                            \
+        promises_builder.append(#promise_name);
+
+        ENUMERATE_PLEDGE_PROMISES
+#undef __ENUMERATE_PLEDGE_PROMISE
+
+        return String::join(" ", promises_builder);
+    }
+
+#define __ENUMERATE_PLEDGE_PROMISE(promise_name) \
+    constexpr Promises& add_##promise_name()     \
```

Comment:
This is, imo, a bit too close to making this possible _and_ trivial:
```c++
Pledge p;
// code
p.add_foo();
// code
p.add_bar();
```

that sounds like the start of something cursed.

Reply:
Yeah the builder was just a POC, I'll get rid of it